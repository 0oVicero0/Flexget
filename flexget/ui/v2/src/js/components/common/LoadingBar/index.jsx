import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import { LinearProgress } from 'material-ui/Progress';

const styleSheet = theme => ({
  root: {
    position: 'relative',
    overflow: 'hidden',
    height: 5,
    backgroundColor: theme.palette.accent[100],
  },
  bar: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    top: 0,
    transition: 'transform 0.2s linear',
    backgroundColor: theme.palette.accent[500],
  },
});


const LoadingBar = ({ loading, classes }) => {
  if (loading) {
    return (
      <LinearProgress
        mode="query"
        classes={{
          root: classes.root,
          bar: classes.bar,
        }}
      />
    );
  }
  return null;
};

LoadingBar.propTypes = {
  loading: PropTypes.bool,
  classes: PropTypes.object.isRequired,
};

LoadingBar.defaultProps = {
  loading: false,
};

export default withStyles(styleSheet)(LoadingBar);
