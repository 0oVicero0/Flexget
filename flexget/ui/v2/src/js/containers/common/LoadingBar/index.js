import { connect } from 'react-redux';
import LoadingBar from 'components/common/LoadingBar';

export function mapStateToProps({ status }, { prefix = '@flexget' } = {}) {
  return {
    loading: !!(Object.keys(status.loading).find(type => type.startsWith(prefix))),
  };
}

export default connect(mapStateToProps)(LoadingBar);
