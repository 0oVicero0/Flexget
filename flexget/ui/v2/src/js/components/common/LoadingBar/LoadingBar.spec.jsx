import React from 'react';
import { shallow } from 'enzyme';
import LoadingBar from 'components/common/LoadingBar';

describe('components/common/LoadingBar', () => {
  it('should render properly when loading', () => {
    expect(
      shallow(<LoadingBar loading />)
    ).toMatchSnapshot();
  });

  it('should render properly when not loading', () => {
    expect(
      shallow(<LoadingBar />)
    ).toMatchSnapshot();
  });
});
