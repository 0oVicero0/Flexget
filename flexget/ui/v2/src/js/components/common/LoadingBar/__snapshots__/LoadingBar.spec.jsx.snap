// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/LoadingBar should render properly when loading 1`] = `
<div
  className="MuiLinearProgress-root-2705929356 LoadingBar-root-1453628708 MuiLinearProgress-rootQuery-1910263825"
>
  <div
    className="MuiLinearProgress-bar-1948415122 LoadingBar-bar-2518219007 MuiLinearProgress-indeterminateBar1-3021942985"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-1948415122 LoadingBar-bar-2518219007 MuiLinearProgress-indeterminateBar2-1048549576"
    style={Object {}}
  />
</div>
`;

exports[`components/common/LoadingBar should render properly when not loading 1`] = `
<div
  className="MuiLinearProgress-root-2705929356 LoadingBar-root-1453628708 MuiLinearProgress-rootQuery-1910263825"
>
  <div
    className="MuiLinearProgress-bar-1948415122 LoadingBar-bar-2518219007 MuiLinearProgress-indeterminateBar1-3021942985"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-1948415122 LoadingBar-bar-2518219007 MuiLinearProgress-indeterminateBar2-1048549576"
    style={Object {}}
  />
</div>
`;
